src/webviews/extension-side/dataviewer/types.ts

```diff
@@ -130,5 +130,6 @@ export interface IJupyterVariableDataProviderFactory {
 
 export const IDataViewerDependencyService = Symbol('IDataViewerDependencyService');
 export interface IDataViewerDependencyService {
-    checkAndInstallMissingDependencies(interpreter: PythonEnvironment): Promise<void>;
+    checkAndInstallMissingDependenciesOnEnvironment(environment: PythonEnvironment): Promise<void>;
+    checkAndInstallMissingDependenciesOnKernel(kernel: IKernel): Promise<void>;
 }
```

Comment:
id prefer to haber a single method, two feels like a work around which introduces debt. 
there's nothing preventing one from using the wrong method.